src/services/NetworkService.js

```diff
@@ -1,17 +1,66 @@
 import Vue from "vue";
+import promiseRetry from "promise-retry";
+import errcode from "err-code";
 
 const AUTH_ROOT = `${process.env.VUE_APP_SERVER_ROOT}/auth`;
 const API_ROOT = `${process.env.VUE_APP_SERVER_ROOT}/api`;
 const ELIGIBILITY_API_ROOT = `${process.env.VUE_APP_SERVER_ROOT}/eligibility`;
 const CONTACT_API_ROOT = `${process.env.VUE_APP_SERVER_ROOT}/contact`;
 
+const FAULT_TOLERANT_HTTP_TIMEOUT = 10000;
+const FAULT_TOLERANT_HTTP_MAX_RETRY_TIMEOUT = 100000;
+const FAULT_TOLERANT_HTTP_MAX_RETRIES = 10;
+
 export default {
   _successHandler(res) {
     return Promise.resolve(res);
   },
   _errorHandler(res) {
     return Promise.reject(res);
   },
+  _faultTolerantHttp(http, method, onRetry, url, data) {
+    return {
+      isAborted: false,
+      retryHttp: function() {
+        return promiseRetry(
+          retry => {
+            if (this.isAborted) {
+              // early exit
+              return Promise.reject(
+                errcode(new Error("Aborted by user"), "EUSERABORTED")
+              );
+            }
+
+            const promise =
+              ["get", "delete", "head", "jsonp"].indexOf(method) !== -1
+                ? http[method](url, {
+                    timeout: FAULT_TOLERANT_HTTP_TIMEOUT
+                  }).then(this._successHandler, this._errorHandler)
```

Comment:
i would generally not handle promise returns and errors nested inside like this (as it can be difficult to follow for future programmers). instead, you could handle this promise outside like:

```
retryHttp: function() {
  return promiseRetry(retry => {
     ...
  }, {
    retries: FAULT_TOLERANT_HTTP_MAX_RETRIES,
    maxTimeout: FAULT_TOLERANT_HTTP_MAX_RETRY_TIMEOUT
  })
    .then(this._successHandler, this._errorHandler)
    .catch(res => {
      if (res.status === 0) { ... }
      ...
    });
}
```

Reply:
The `catch` handler needs to be nested inside the function that is passed to `promiseRetry` because it needs to call the passed-in `retry` function to signal to `promiseRetry` that the function needs to be retried.

I think I'm going to use a different package for managing the retries and backoffs and rewrite this whole chunk of code--the `promiseRetry` module looked tempting but as I started implementing the things I needed to implement here (like stopping the requests when the user clicks "Abort") my solutions became increasingly ugly.